networkx/algorithms/tree/distance_measures.py

```diff
@@ -0,0 +1,125 @@
+import networkx as nx
+
+__all__ = [
+    "tree_centroid",
+]
+
+
+def _subtree_sizes(T, root):
+    """Return a `dict` of subtree sizes in a tree rooted at a given node.
+
+    Parameters
+    ----------
+    T : NetworkX graph
+       A tree.
+
+    root : node
+       A node in `T`.
+
+    Returns
+    -------
+    s : dict
+       Dictionary of subtree sizes keyed on nodes.
+
+    Examples
+    --------
+    >>> _subtree_sizes(nx.path_graph(4), 0)
+    {0: 4, 1: 3, 2: 2, 3: 1}
+
+    >>> _subtree_sizes(nx.path_graph(4), 2)
+    {2: 4, 1: 2, 0: 1, 3: 1}
+    """
+    sizes = {root: 1}
+    stack = [root]
+    for a, b in nx.dfs_edges(T, root):
+        while stack[-1] != a:
+            x = stack.pop()
+            sizes[stack[-1]] += sizes[x]
+        stack.append(b)
+        sizes[b] = 1
+    for b, a in nx.utils.pairwise(reversed(stack)):
+        sizes[a] += sizes[b]
+    return sizes
+
+
+@nx.utils.not_implemented_for("directed")
+@nx._dispatchable
+def tree_centroid(T):
```

Comment:
The dispatching test is failing because it is looking for input `G` and the input is named `T`.
You can either:
- add an argument to the dispatchable decorator giving the name of the graph.
- change the name of the input to `G`.

All your examples seem to use `G` as the graphs name anyway, so maybe it is best to change the `T`s to `G`s.